### PR TITLE
Holotool for the RD

### PR DESCRIPTION
**Pull request by ghost**
# STOP FUCKING ASKING IF THEY VIBRATE

Holotools!  sprites from /vg/
capture

:cl: More Robust Than You
add: Experimental Holotools. The RD spawns with one in their locker. They do not vibrate, stop asking.
add: SERIOUSLY, THEY DO NOT VIBRATE.
/:cl:

They can switch into high speed tools (not as fast as CE tools, but faster than normal). also they're traitor objectives. and you can change their colors.

### CODE REVIEW

**Comment by FantasticFwoosh** [comment]
Do they have a _vibrate_ setting?

**Comment by MoreRobustThanYou** [comment]
no

**Comment by DoctoreMad** [comment]
​>See this actually useful item you could potentially give to your fellow co-workers like a head of staff is supposed to do?
​>Its a traitor objective

**Comment by MoreRobustThanYou** [comment]
@DoctoreMad so?

**Comment by InsaneHyena** [comment]
What about the vibrate setting? Does the holotool come with that?

**Comment by MoreRobustThanYou** [comment]
@InsaneHyena NO IT DOES NOT VIBRATE

**Comment by DoctoreMad** [comment]
@MoreRobustThanYou "So?" So SO I CAN'T WAKE UP. THERE ARE 3 PEOPLE IN YOUR PR AND NONE OF THEM LIKE YOUR TOOL.
VIBRATE MODE OR REMOVE ANTAG OBJECTIVE
THIS IS YOUR 1 SHOT AT GLORY

**Comment by MoreRobustThanYou** [comment]
@DoctoreMad no erp

File: code/game/objects/items/holotool.dm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	AddTool(/obj/item/device/multitool, "holo-multitool")
+
+/obj/item/holotool/Initialize()
+	. = ..()
+	//create and rename tools
+	AddTools()
+
+/obj/item/holotool/attack_self(mob/living/user)
+	to_chat(user, "<span class='notice'>You turn [src] [on ? "off" : "on"].</span>")
+	on = !on
```
Comment (ChangelingRain) [inline code/game/objects/items/holotool.dm]: why do you even have this toggle.

**Comment by ChangelingRain** [comment]
Wow jeez those look awful

**Comment by DoctoreMad** [comment]
What does Vibration have to do with ERP? I a puzzled really. I was more imagining a effect with the holotools revving ominously like His Grace to scare of predators.

**Comment by PraiseRatvar** [comment]
i told you on coderbus that this isn't a good idea yesterday, i think @pudl said so as well.

why didn't you lisssteeeeen.

**Comment by GunHog** [comment]
I have always wanted a pink multi-tool for cyborg 'maintenance'...

**Comment by Anonmare** [comment]
Okay so they don't have a vibrate setting. But do they have a "electro-stimulation" setting?

**Comment by FantasticFwoosh** [comment]
Please get some new sprites, it looks like something you buy from a adult toy shop made out of those cheap plastics.

File: code/game/objects/items/holotool.dm (hunk 1 above)
Comment (Militaires) [reply code/game/objects/items/holotool.dm]: for the chat message

**Comment by BeeSting12** [comment]
they look like dildos with tools attached.

File: code/game/objects/items/holotool.dm (hunk 1 above)
Comment (ChangelingRain) [reply code/game/objects/items/holotool.dm]: that doesn't answer why the toggle exists at all.

**Comment by ExcessiveUseOfCobblestone** [comment]
RD's tools should not be better than that of the actual [de]construction crew wtf?

In all honesty it should be worse but be a single device you can switch between to keep a thematic sense of why it has value for antagonists.

**Comment by FantasticFwoosh** [comment]
Ayy tools are better

**Comment by KingPhilipIII** [comment]
But the question remains. Does it vibrate if you use another multitool on it? Secret functions?

**Comment by MoreRobustThanYou** [comment]
if someone else is willing to sprite these, tell me

**Comment by factoryman942** [comment]
it's not a vibrator, it's a **back massager**

**Comment by kevinz000** [comment]
mrty good pr but if you fuck this up and ruin it forever i swear to fucking god.

**Comment by kevinz000** [comment]
also
WTF I LOVE MRTY NOW?!!
(ill rescind this when he breaks either this idea or the entire codebase 5 minutes later.)

**Comment by kevinz000** [comment]
also jesus christ use vg's sprites they're 2^32x better

**Comment by GunHog** [comment]
Original VG sprite PR images: https://github.com/vgstation-coders/vgstation13/pull/15077

VG

Latest VG sprites: https://github.com/vgstation-coders/vgstation13/pull/15830
VG NEW

**Comment by JamesDenholm** [comment]
And most importantly, it doesn't look like a sex toy.

At least, not as much.

**Comment by ChangelingRain** [comment]
the holoshield inhand looks like shit tho but since this is tools only that wouldn't matter

**Comment by meeeeme** [comment]
But can you stick your penis in it?

**Comment by optimumtact** [comment]
Science buff for no reason 👎

**Comment by optimumtact** [comment]
also steps over CE's stuff, that's a hard negative imo

**Comment by JamesDenholm** [comment]
that's because science is strictly better than engineering even if graduate salaries would suggest otherwise

More seriously, it's slower than the CE tools -- if something can't exist comfortably between CE and regular tools, maybe CE tools need to be turned up to sanic speed

**Comment by optimumtact** [comment]
no

**Comment by JamesDenholm** [comment]
So surely then it's fine for something to exist in the middle.

**Comment by FantasticFwoosh** [comment]
We already have that middle ground by the ability to produce tools using ayy tech that are twice as fast (requires studying something of ayy tech origin) as well as making CE tools already using high tech.

- Ideally given how powercrept this is, it should have a exceedingly high materials tech cost of 8 or something because you're literally shaping super advanced holo-refraction technology.

**Comment by AnonymousNow** [comment]
Seems cute. Not sure about balance, but I think it's something that could be worked into the game somewhere.

**Comment by JamesDenholm** [comment]
If powercreep is an issue, it could feasibly be, like, slower than regular tools? Thus it's not all upside.

**Comment by XDTM** [comment]
Should be slightly slower than basic tools, the upside is that it fits in a pocket

File: code/game/objects/items/holotool.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	var/on = FALSE
+	var/obj/item/current_tool = 0
+	var/current_color = "#48D1CC" //mediumturquoise
+
+/obj/item/holotool/proc/AddTool(typee, namee)
+	var/obj/item/WR = new typee(src)
+	WR.forceMove(src)
+	WR.name = namee
+	WR.usesound = usesound //use the same sound as we do
+	WR.toolspeed = 0.5
```
Comment (Arianya) [inline code/game/objects/items/holotool.dm]: Would change this to 1.25 to alleviate powercreep concerns, its a huge space saving compared to a normal set of tools.

**Review by Arianya — CHANGES_REQUESTED** [review]

**Comment by GunHog** [comment]
The internal welder starts turned off, and there is no way to toggle it on to use or off again to refuel. Also, the on/off mode of the holotool is pointless. It already has a "nothing" state which is evident when it is toggled "on" without a tool selected. 

You can solve both issues by passing the `attack_self()` to the selected tool itself, and removing the on/off mode entirely. 

Make new tool subtypes so you do not have to modify the internal tools at `Initialize()`. This will also let you give the tools their own descriptions (passing the `examine()` to the tool as you do with `attack_self()`. This is important for the welder to let the user check its fuel level. Do not forget to set NODROP on the tools to prevent removal bugs. 

Another recommendation is to have them produce very weak light in the color of the tool, as it is made of light, after all!

Btw, please do not make this _worse_ than basic tools. The space saving is not as important for an RD in the beginning, because the best use for his belt slot is a toolbelt (which most RDs would grab from Robotics). The only other use for the RD's belt slot, in my personal experience, in the beginnings of the round would be a Security belt (during Revolution) or his extended baton if he needs the suit slot to hold an air tank. 

If you do make it slower than regular tools, remove it as a Traitor objective, as an RD has little incentive to protect the holotool. I used to happily give away my reactive armor (when it had a slowdown and lower activation chance, making it mostly unused) in exchange for Traitor items or just my life. I still do sometimes sell my armor, but as it is valuable, I am more inclined to negotiate for a better deal from the Traitor, rather than just giving it up for basically nothing.

**Comment by nicbn** [comment]
Make it the same speed as common tools and port vg sprites.

**Comment by nicbn** [comment]
Also change them to greyscale.

**Comment by GunHog** [comment]
This is tools.dmi as a .png file as offered by the PR. They *are* greyscale.

tools

**Comment by Repukan** [comment]
Should be slower than normal tools since it's a small pocket fitting tool belt.

**Comment by MoreRobustThanYou** [comment]
> Make new tool subtypes so you do not have to modify the internal tools at Initialize(). This will also let you give the tools their own descriptions (passing the examine() to the tool as you do with attack_self() ). This is important for the welder to let the user check its fuel level. Do not forget to set NODROP on the tools to prevent removal bugs.

To be honest, I don't like doing this.

**Comment by GunHog** [comment]
Why not? It is better performance wise (The variables are already set), as well as not having your tools broken if someone changes the originals. It is to your advantage.

**Comment by MoreRobustThanYou** [comment]
New sprites!

**Comment by Repukan** [comment]
Can you add the vibration though

**Comment by MoreRobustThanYou** [comment]
NO

**Comment by Repukan** [comment]
😢

**Comment by DoctoreMad** [comment]
Can you add a vibration mode to the Research Director then

**Comment by DoctoreMad** [comment]
He's a massive dildo on most rounds so that wouldn't change a lot anyways

**Comment by JamesDenholm** [comment]
Remember when people gave commits useful names?

Yeah me neither

**Comment by MoreRobustThanYou** [comment]
@JamesDenholm you don't know me, i usually give commits names based on what i'm thinking

**Comment by JamesDenholm** [comment]
This explains a lot

**Comment by ChangelingRain** [comment]
Why do you keep greyscaling things and making them look awful?

**Comment by Anonmare** [comment]
**DO 👏 THEY 👏 VIBRATE???**

**Comment by lzimann** [comment]
Regardless of sprite being good or not, the whole concept of one slot(one item) for every tool is just plain stupid. Unless @KorPhaeron or someone else really wants this, I'll just hit close so we can all move on without this kind of stuff.

**Comment by kevinz000** [comment]
lmao btfo.

**Comment by GunHog** [comment]
To be expected, I guess. It is the same place that will not even allow the RD to start with a toolbelt...

**Comment by Repukan** [comment]
Oh poor RD player his powercreep won't get merged, boo fucking hoo woe is me.

**Comment by MoreRobustThanYou** [comment]
@Repukan fuck off

**Comment by Repukan** [comment]
@MoreRobustThanYou no, fuck you, Leatherman.

**Comment by JamesDenholm** [comment]
It's absolutely trivial for an RD to get a toolbelt, toolbox, and multitool, I don't see the distinction.

**Comment by AnonymousNow** [comment]
Like I said, I like the concept and think it should be in the game, but I don't think it works as a unique roundstart RD item. Make it accessible around the same time on the tech tree as toolset arm implants.

**Comment by kevinz000** [comment]
@Repukan @MoreRobustThanYou go fight each other in the thunderdome and show who's the boss of this github :^)

**Comment by Repukan** [comment]
Pretty sure we're both permabanned

**Comment by GunHog** [comment]
@lzimann and @KorPhaeron I would like to try convincing you one last time.

Why do you feel that having one item having each tool is stupid? If you made this holotool be a normal sized item where it does not fit in pockets, then it no longer is an advantage.

So, if it is normal size, then it becomes very similar to holding your toolbelt in your hand. With a toolbelt, you can withdraw the tools you need, and have one tool in each hand for faster interaction for multi-step things such as building mechs and hacking doors, all without actually removing the belt from its slot.

The holotool, however, requires the user to hold it, and only one tool can be active at a time. In common scenarios such as being slipped or shocked, the user will drop at most two of his tools, and keep the toolbelt. If you drop your holotool, you _lose all of your tools at once._ It hurts just that little bit more when a greyscum makes off with it. 

This is good for a minor increase in conflict as well, for that reason. The tool remains valuable thanks to the upgrade ability gained from engineering tech disks. It remains inferior to CE tools at all times, as those are 4x speed, the holotool caps at 2x with Engineering 6.

**TL;DR:** It is not OP if you make it normal sized.

Thank you for your consideration.

**Comment by Repukan** [comment]
Rito pls buffe my rol reeeeeeeeeeeeeeeee

**Comment by nicbn** [comment]
> This is tools.dmi as a .png file as offered by the PR. They are greyscale.

The /vg/ ones I mean.
Also I want holotools don't close this reee.

**Comment by DrSnips** [comment]
I'm real late to the party, but I came here to laugh. /vg/ has had zero issues with balance of the holotool since its addition. Also the reason I never used greyscale for colorchange is because it looks ugly as sin.

**Comment by Repukan** [comment]
You mean a forced mid rp server with different coding standards and directions has a different outcome to items added?

Wow, imagine my fucking shock dude.

**Comment by DrSnips** [comment]
I'm not here to stir shit I'm just pointing things out. Standards and playstyle isn't something I care about here. I'm laughing because of the sprites mainly and how this side pretty much made it into a joke about dildos.

**Comment by Repukan** [comment]
It did look like a dildo.

**Comment by DrSnips** [comment]
I think the colors killed it the most here in my opinion. Ah well I'm done laughing here.

**Comment by GunHog** [comment]
Another sprite set was added which made them look better! There was not a significant balance implication with the item, just that Izimann was against it being multiple tools in one item. I am sure someone would ask me if my holotool acts as a vibrator ingame, though. It would have been BOUND to happen. There was minor ERP potential with borgs too. "Magni Bronzebeard carefully slides his vibrating holotool into Engineering Cyborg-239's waiting entrance."

**Comment by DrSnips** [comment]
While I admit the new tools that were made look nice they just seem a bit out of proportion to the tool's handle. When I made them I based the icon sizes around the existing switchtool sprites. But hey this PR is  dead and I heard someone tried to port the holotool I made so I thought I'd drop by. So I'm going to leave it at that.

**Comment by AnonymousNow** [comment]
Yeah, but I want to make sure that we're all certain on the most important point:

Does it vibrate?
